test/rrule.test.ts

```diff
@@ -3716,4 +3704,63 @@ describe('RRule', function () {
     const rule = new RRule(options)
     expect(rule.toText()).to.equal('every day')
   })
+
+  describe('time zones', () => {
+    const targetZone = 'America/Los_Angeles'
+    const startDate = DateTime.utc(2013, 8, 6, 11, 0, 0)
+    const targetOffset = startDate.setZone(targetZone).offset
+    const dtstart = startDate.toJSDate()
+
+    it('generates correct recurrences when recurrence is in dst and current time is standard time', () => {
+      const local = DateTime.local(2013, 2, 6, 11, 0, 0)
+      setMockDate(local.toJSDate())
+
+      const rule = new RRule({
+        dtstart,
+        count: 1,
+        tzid: targetZone
+      })
+      const recurrence = rule.all()[0]
+      const expected = expectedDate(startDate, local)
+
+      expect(recurrence)
+        .to.deep.equal(
+          expected 
+        )
+
+      resetMockDate()
+    })
+
+  it('generates correct recurrences when recurrence is in dst and current time is dst', () => {
+      const local = DateTime.local(2013, 8, 6, 11, 0, 0)
+      setMockDate(local.toJSDate())
+
+      const rule = new RRule({
+        dtstart,
+        count: 1,
+        tzid: targetZone
+      })
+      const recurrence = rule.all()[0]
+      const expected = expectedDate(startDate, local)
+
+      expect(recurrence)
+        .to.deep.equal(
+          expected 
+        )
+
+      resetMockDate()
+    })
+
+    function expectedDate(startDate: DateTime, local: DateTime): Date {
+      const { zoneName: systemZone } = local
+      const {
+        offset: systemOffset,
+      } = startDate.setZone(systemZone)
+
+      const netOffset = targetOffset - systemOffset
+      const hours = -((netOffset / 60) % 24)
+      const minutes = -(netOffset % 60)
+      return DateTime.fromJSDate(dtstart).plus({ hours, minutes }).toJSDate()
```

Comment:
This would be the same as doing `return startDate.plus({ hours, minutes }).toJSDate()` and less confusing.  At first glance the question would be where `dtstart` comes from but it's inside the `describe` block